tileview/src/main/java/com/qozix/tileview/widgets/ZoomPanLayout.java

```diff
@@ -598,19 +608,19 @@ protected int getConstrainedScrollY( int y ) {
   }
 
   protected int getScrollLimitX() {
-    return mScaledWidth - getWidth();
+    return mScaledWidth - getWidth() + FloatMathHelper.scale(mExtraBorder, mScale);
```

Comment:
let's precompute these values.  i'd probably do something like have `mExtraBorder` and `mScaledExtraBorder`.  whenever either value changes (scale or the border) recompute it and save the value in the latter variable.  that way each of these 4 methods (`getScrollLimitX`, `getScrollLimitY`, `getScrollMinX `, `getScrollMinY `) do not need to recompute every time and can just reference the `mScaledExtraBorder` value (or whatever name that turns out to be, in consideration of other comments)

Reply:
Sounds like a good idea. I'll fix that.

How about `getConstrainedScrollX` and `getConstrainedScrollY`? Couldn't those have more precomputing as well with the same reasoning? (though that is outside the scope of these changes)

Reply:
That's a good point.  TBH it's been many years since that was written so it might have just been an oversight or laziness, or maybe it's because there's so many contributing variables that it seemed easier to compute on the fly...  this library is old and rickety, i'll admit (thus, V3)